src/controllers/userController.js

```diff
@@ -41,3 +41,58 @@ export const editUserRole = async (req, res) => {
     return respondWithWarning(res, 500, 'Server Error');
   }
 };
+
+export const createProfile = async (req, res) => {
+  // de-structure id out of req.auth
+  const { id } = req.auth;
+  // Add the de-structure id to req.body with a new key of userId for ease of storage.
+  req.body.userId = id;
```

Comment:
Since when the user is creating their profile, i am not requiring that they provide there ```id``` . Order than through the token passed from the header, for the userId to be part of the object am storing in the database like 
```export const createProfile = (profile) => Model.Profile.create(profile); ``` that is why i added to the ```req.body``` 
object representing the object to be persisted.

Reply:
Since when the user is creating their profile, i am not requiring that they provide there ```id``` . Order than through the token passed from the header, for the userId to be part of the object am storing in the database like 
```export const createProfile = (profile) => Model.Profile.create(profile); ``` that is why i added to the ```req.body``` 
object representing the object to be persisted.